In [8]:
import os
os.environ["CALITP_BQ_MAX_BYTES"] = str(100_000_000_000)

from calitp.tables import tbl
from calitp import query_sql
import calitp.magics
import branca

import shared_utils
import utils

from siuba import *
import pandas as pd
import geopandas as gpd
import shapely

import datetime as dt
import time
from zoneinfo import ZoneInfo

import rt_analysis2 as rt
import importlib

importlib.reload(utils)
importlib.reload(rt)

import gcsfs
fs = gcsfs.GCSFileSystem()

In [2]:
# analysis_date = dt.date(2022, 1, 11)
## all data incomplete -- did it crash? (nothing before 9ish)

In [4]:
analysis_date = dt.date(2022, 1, 12)

In [5]:
samtrans_itp_id = 290

In [4]:
# utils.get_vehicle_positions(samtrans_itp_id, analysis_date)

In [5]:
# utils.get_trips(samtrans_itp_id, analysis_date)

In [6]:
# utils.get_stop_times(samtrans_itp_id, analysis_date, force_clear=False)

In [7]:
# utils.get_stops(samtrans_itp_id, analysis_date)

In [8]:
# sam_trips >> head(5)

In [9]:
# sam_vp >> head(5)

In [10]:
# sam_vp >> inner_join(_, sam_trips, on ='trip_id')

In [11]:
## TODO
# lbt_analysis.position_interpolators['9288623']['rt'].trip_key
# generated 1553149201202261125 on 11/30/21 but this doesn't seem to exist?

In [6]:
air_joined = pd.read_parquet('airtable_joined.parquet')

In [7]:
air_joined

Name  calitp_itp_id  \
0                                  Long Beach Transit            170   
1               Peninsula Corridor Joint Powers Board            246   
2                                     City of Visalia            361   
3                   San Mateo County Transit District            290   
4                       Marin County Transit District            194   
5                            Monterey-Salinas Transit            208   
6                Napa Valley Transportation Authority            218   
7   Golden Gate Bridge, Highway and Transportation...            127   
8    Transit Joint Powers Authority for Merced County            343   
9                Sacramento Regional Transit District            273   
10                Yolo County Transportation District            372   
11                     Golden Empire Transit District            126   
12                                     City of Madera            188   
13            Kings County Area Public Transit Agency            148   
14                         Tulare County Area Transit            346   
15           Ventura County Transportation Commission            380   
16                               City of Santa Monica            300   
17                                City of Los Angeles            183   
18  Los Angeles County Metropolitan Transportation...            182   
19                                   City of Pasadena            243   
20                              City of Santa Clarita            295   
21                                    City of Burbank             45   
22                              City of Redondo Beach            260   
23                                   City of Commerce             75   
24                              Solano County Transit            310   
25                                      Sonoma County            314   
26                                   City of Petaluma            247   
27            Sonoma-Marin Area Rail Transit District            315   
28                                 City of Santa Rosa            301   
29     Livermore / Amador\n  Valley Transit Authority            167   
30        Santa Clara Valley Transportation Authority            294   
31             Eastern Contra Costa Transit Authority            336   
32                   City and County of San Francisco            282   
33                                 City of Union City            350   
34                                  City of Fairfield            110   
35                                    City of Modesto            203   
36                                    City of Turlock            349   
37                                      Nevada County            221   
38             Orange County Transportation Authority            235   
39                    Victor Valley Transit Authority            360   
40                           Riverside Transit Agency            269   
41                               Banning Pass Transit             30   
42                             Lake Transit Authority            159   
43                         Humboldt Transit Authority            135   
44        Santa Barbara Metropolitan Transit District            293   
45                   Eastern Sierra Transit Authority             99   
46                      North County Transit District            226   
47                         Redding Area Bus Authority            259   

      Reporting Category     Caltrans District GTFS Static Status  \
0                   Core      07 - Los Angeles  Static Incomplete   
1                   Core          04 - Oakland  Static Incomplete   
2                   Core           06 - Fresno  Static Incomplete   
3                   Core          04 - Oakland  Static Incomplete   
4                   Core          04 - Oakland  Static Incomplete   
5                   Core  05 - San Luis Obispo  Static Incomplete   
6                   Core          04 

In [8]:
for agency in air_joined.calitp_itp_id:
    try:
        analysis = rt.OperatorDayAnalysis(agency, analysis_date)
    except Exception as e:
        print(f'failed on agency {agency}')
        print(e)

found parquet
found parquet
found parquet
found parquet
found_parquet
9289275
sc_called
check location data for trip -4329841926259451152
sc_called
(39, 12)
9288197
sc_called
check location data for trip 477904054299874101
sc_called
(70, 12)
check location data for trip 477904054299874101
sc_called
(65, 12)
check location data for trip 477904054299874101
sc_called
(63, 12)
9288671
sc_called
9288672
sc_called
check location data for trip -4332510122068877721
sc_called
(73, 12)
9288276
sc_called
check location data for trip -8407187321077433825
sc_called
(47, 12)
check location data for trip -8407187321077433825
sc_called
(43, 12)


/opt/conda/lib/python3.9/site-packages/siuba/sql/utils.py:52: SAWarning: Dialect bigquery:bigquery will not make use of SQL compilation caching as it does not set the 'supports_statement_cache' attribute to ``True``.  This can have significant performance implications including some performance degradations in comparison to prior SQLAlchemy versions.  Dialect maintainers should seek to set this attribute to True after appropriate development and testing for SQLAlchemy 1.4 caching support.   Alternatively, this attribute may be set to False which will disable this warning. (Background on this error at: https://sqlalche.me/e/14/cprf)
  return self.connectable.execute(*args, **kwargs)


gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_246_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 246
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_361_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 361
routelines must not be empty
found parquet
found parquet
found parquet
found parquet
found_parquet
11031698-136-Blocks-Weekday-54
sc_called
check location data for trip -1949207877340033680
sc_called
(139, 12)
check location data for trip -1949207877340033680
sc_called
(136, 12)
check location data for trip -1949207877340033680
sc_called
(134, 12)
check location data for trip -1949207877340033680
sc_called
(132, 12)
check location data for trip -1949207877340033680
sc_called
(130, 12)
check location data for trip -1949207877340033680
sc_called
(128, 12)
check location data for trip -1949207877340033680
sc_called
(126, 12)
check location data for trip -1949207877340033680
sc_called
(124, 12)
check location data for trip -1949207877340033680
sc_called
(122, 12)
check location data for trip -1949207877340033680
sc_called
(120, 12)
check location data for trip -1949207877340033680
sc_called
(118, 12)
check location data for trip -1949207877340033680
sc_called
(116, 12)

/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 194
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_208_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")


failed on agency 208
vehicle positions trip ids not in schedule
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_218_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 218
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_127_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 127
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_343_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")


failed on agency 343
vehicle positions trip ids not in schedule
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_273_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")


failed on agency 273
vehicle positions trip ids not in schedule
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_372_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")


failed on agency 372
vehicle positions trip ids not in schedule
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_126_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 126
routelines must not be empty


/home/jovyan/data-analyses/rt_delay/utils.py:140: UserWarning: rt data starts after analysis date
  warnings.warn('rt data starts after analysis date')


gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_188_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 188
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_148_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 148
routelines must not be empty
found parquet
found parquet
found parquet
found parquet
failed on agency 346
vehicle positions trip ids not in schedule
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_380_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")


failed on agency 380
vehicle positions trip ids not in schedule
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_300_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 300
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_183_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 183
routelines must not be empty
found parquet
found parquet
found parquet
found parquet
found_parquet
10105002581045-DEC21
sc_called
10053002331050-DEC21
sc_called
10051003301041-DEC21
sc_called
None
vehicle positions gdf must not be empty
10055002841003-DEC21
sc_called
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_243_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 243
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_295_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 295
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_45_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 45
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_260_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 260
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_75_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 75
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_310_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 310
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_314_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 314
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_247_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 247
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_315_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 315
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_301_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 301
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_167_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 167
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_294_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 294
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_336_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 336
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_282_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 282
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_350_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 350
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_110_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 110
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_203_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 203
routelines must not be empty


/home/jovyan/data-analyses/rt_delay/utils.py:140: UserWarning: rt data starts after analysis date
  warnings.warn('rt data starts after analysis date')


gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_349_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 349
routelines must not be empty


/home/jovyan/data-analyses/rt_delay/utils.py:140: UserWarning: rt data starts after analysis date
  warnings.warn('rt data starts after analysis date')


gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_221_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 221
routelines must not be empty


/home/jovyan/data-analyses/rt_delay/utils.py:140: UserWarning: rt data starts after analysis date
  warnings.warn('rt data starts after analysis date')


gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_235_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 235
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_360_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")


failed on agency 360
vehicle positions trip ids not in schedule
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_269_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 269
routelines must not be empty


/home/jovyan/data-analyses/rt_delay/utils.py:140: UserWarning: rt data starts after analysis date
  warnings.warn('rt data starts after analysis date')


gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_30_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 30
routelines must not be empty


/home/jovyan/data-analyses/rt_delay/utils.py:140: UserWarning: rt data starts after analysis date
  warnings.warn('rt data starts after analysis date')


gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_159_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 159
routelines must not be empty


/home/jovyan/data-analyses/rt_delay/utils.py:140: UserWarning: rt data starts after analysis date
  warnings.warn('rt data starts after analysis date')


gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_135_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 135
routelines must not be empty
gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_293_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 293
routelines must not be empty


/home/jovyan/data-analyses/rt_delay/utils.py:140: UserWarning: rt data starts after analysis date
  warnings.warn('rt data starts after analysis date')


gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_99_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 99
routelines must not be empty


/home/jovyan/data-analyses/rt_delay/utils.py:140: UserWarning: rt data starts after analysis date
  warnings.warn('rt data starts after analysis date')


gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_226_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 226
routelines must not be empty


/home/jovyan/data-analyses/rt_delay/utils.py:140: UserWarning: rt data starts after analysis date
  warnings.warn('rt data starts after analysis date')


gs://calitp-analytics-data/data-analyses/rt_delay/cached_views/
stops_259_2022-01-12


/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises.  We do not yet recommend using this in a production setting unless you are able to rewrite your Parquet/Feather files.

To further ignore this warning, you can do: 
import warnings; warnings.filterwarnings('ignore', message='.*initial implementation of Parquet.*')
  gdf.to_parquet(f"./{FILE_NAME}.parquet")
/home/jovyan/data-analyses/_shared_utils/shared_utils/utils.py:36: UserWarning: this is an initial implementation of Parquet/Feather file support and associated metadata.  This is tracking version 0.1.0 of the metadata specification at https://github.com/geopandas/geo-arrow-spec

This metadata specification does not yet make stability promises

failed on agency 259
routelines must not be empty


In [ ]:
## fix routelines thing...

In [26]:
# metro_analysis.position_interpolators['10105002581045-DEC21']['rt'].detailed_speed_map()

In [27]:
# lbt_analysis.position_interpolators['9288672']['rt'].detailed_speed_map()

In [6]:
sam_analysis = rt.OperatorDayAnalysis(samtrans_itp_id, analysis_date)

found parquet
found parquet
found parquet
found parquet
found_parquet
11031698-136-Blocks-Weekday-54
sc_called
check location data for trip -1949207877340033680
sc_called
(139, 12)
check location data for trip -1949207877340033680
sc_called
(136, 12)
check location data for trip -1949207877340033680
sc_called
(134, 12)
check location data for trip -1949207877340033680
sc_called
(132, 12)
check location data for trip -1949207877340033680
sc_called
(130, 12)
check location data for trip -1949207877340033680
sc_called
(128, 12)
check location data for trip -1949207877340033680
sc_called
(126, 12)
check location data for trip -1949207877340033680
sc_called
(124, 12)
check location data for trip -1949207877340033680
sc_called
(122, 12)
check location data for trip -1949207877340033680
sc_called
(120, 12)
check location data for trip -1949207877340033680
sc_called
(118, 12)
check location data for trip -1949207877340033680
sc_called
(116, 12)
check location data for trip -1949207877340033680

In [ ]:
rt.OperatorDayAnalysis._generate_stop_delay_view(sam_analysis)

insufficient bounding points for trip -4321367742262557686, location 0.0: start/end of route?
insufficient bounding points for trip -4321367742262557686, location 363.46020677282127: start/end of route?


/opt/conda/lib/python3.9/site-packages/geopandas/geodataframe.py:1351: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  super().__setitem__(key, value)


In [45]:
## TODO '11031623-136-Blocks-Weekday-54' should not be eastbound...

In [198]:
# example_vp_trip.detailed_speed_map()

## To-do online

* test new getters
* verify info columns, drop and add to joins as needed
* investigate 294 bug, time-permitting
* ...
* run for all existing operators... (reality check results...)